lib/galaxy/workflow/modules.py

```diff
@@ -598,6 +598,8 @@ def _find_collections_to_match(self, progress: "WorkflowProgress", step, all_inp
                     effective_input_collection_type,
                     dataset_collection_type_descriptions,
                 )
+                if history_query.direct_match(data):
```

Comment:
This might break mapped over conditional subworkflow steps, no ? It feels like something done deliberately, see the comment above, from line 590.

Reply:
You are right and I think there was a test in CI for that (test_invocation_map_over) that was red. I've pushed a workaround that doesn't do this if there is a subworkflow - but my guess is we should be handling these two conditions in parallel better. I'll see if I can write a variant of my test case that maps a list:list:list over the tool that takes in list or list:list. We're in agreement about the nature of the bug though right? My test case makes sense - running a list:list of a tool with a list|list:list input should not result in mapping and the extension running the list:list:list over that tool should result in a single level of mapping and not two?

Reply:
yes, I think I had just not considered that case before.